compass-js-frontend/src/main/webapp/model/scenenode.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of COMPASS. It is subject to the license terms in
+ * the LICENSE file found in the top-level directory of this distribution.
+ * (Also available at http://www.apache.org/licenses/LICENSE-2.0.txt)
+ * You may not use this file except in compliance with the License.
+ */
+
+"use strict";
+
+var AmpersandModel = require("ampersand-model");
+var Config = require('../config');
+var ComponentCollection = require("../collection/component-collection");
+
+var SceneNode = AmpersandModel.extend({
+	props: {
+		id: "number",
+		name: {
+			type: "string",
+			required: true,
+			default: "Scene Node"
+		},
+		children: {
+			childNodes: SceneNode
+		},
+		parent: "number",
+		selectable3d: "boolean",
+		visible: "boolean"
+	},
+	collections: {
+		components: ComponentCollection
+	},
+	ajaxConfig: {
+		headers: {
+			"Accept": "application/json"
+		}
+	},
+	init: function() {
+		console.log("initializing SceneNode");
+		this.childNodes = new Array();
+	},
+	parse: function(attrs){
```

Comment:
I'm fairly certain this can be made more concise.
Am I right in assuming that most of this function's purpose is getting the "children" property into the right shape?